backend/clustering_runner.py

```
@@ -0,0 +1,233 @@
+        created_at=now,
+        stats={},
+    )
+    add_cluster_job(job)
+
+    locked = acquire_cluster_lock(project_id, job_id, ttl_seconds=600)
+    if not locked:
+        logger.info("Clustering already running for project %s", project_id)
+        return job
+
+    loop = asyncio.get_event_loop()
+    loop.create_task(run_clustering_job(project_id, job_id))
```

Ruff (0.14.8)

Store a reference to the return value of `loop.create_task`

(RUF006)